lib/openapi_parser.rb

```diff
@@ -13,14 +16,42 @@
 
 module OpenAPIParser
   class << self
+    # Load schema yaml object. Uri is not set for returned schema.
     # @return [OpenAPIParser::Schemas::OpenAPI]
     def parse(schema, config = {})
-      c = Config.new(config)
-      root = Schemas::OpenAPI.new(schema, c)
+      load_yaml(schema, config: Config.new(config), uri: nil, schema_registry: {})
+    end
 
-      OpenAPIParser::ReferenceExpander.expand(root) if c.expand_reference
+    # Load schema in specified filepath. If file path is relative, it is resolved using working directory.
+    # @return [OpenAPIParser::Schemas::OpenAPI]
+    def load(filepath, config = {})
+      path = Pathname.new(filepath)
+      path = Pathname.getwd + path if path.relative?
+      load_uri(URI::File.build(path: path.to_s), config: Config.new(config), schema_registry: {})
+    end
+
+    # Load schema located by the passed uri. Uri must be absolute.
+    # @return [OpenAPIParser::Schemas::OpenAPI]
+    def load_uri(uri, config:, schema_registry:)
+      # Open-uri doesn't open file scheme uri, so we try to open file path directly
+      # File scheme uri which points to a remote file is not supported.
+      content = if uri.scheme == 'file'
+        open(uri.path, &:read)
+      else
+        uri.open(&:read)
+      end
 
-      root
+      load_yaml(YAML.load(content), config: config, uri: uri, schema_registry: schema_registry)
```

Comment:
I think we mustn't trust that definition files downloaded from outside.   
And OpenAPI definition does not allow serialized ruby object.  
So we should use Psych.safe_load  
https://ruby-doc.org/stdlib-2.7.0/libdoc/psych/rdoc/Psych.html#safe_load-method

Reply:
The OpenAPI definition support YAML and JSON file.
So we should support load YAML and JSON file.
(Until now, this library had received the loaded hash, so there was no problem.)

I think it is good to identify by Content-Type of the header, but I think it is good to identify by extension because it is difficult.
When we support other http options (like proxy, cookie, etc...), we'll add support Content-Type :)

Reply:
@ota42y I had completely forgotten JSON version 😢 

Now `load_uri` checks the resources's extension and decides the format(json or yaml). If there is no file extension, or unknown file extension, it tries both formats.